### PR TITLE
Add support for screen sharing in 1:1 calls

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes [#12652](https://github.com/vector-im/element-web/issues/12652)
**Requires [#1685](https://github.com/matrix-org/matrix-js-sdk/pull/1685)**

<hr></hr>

Screenshot_20210604_082501

This removes the shift-click screen-sharing because it could lead to problems and it _always_ combines user-media and screen-sharing, even if it is not necessary.

<hr></hr>

Screenshot_20210716_212921

### CODE REVIEW

File: src/components/views/rooms/RoomHeader.js [hunk 1]
```diff
@@ -117,6 +119,14 @@ export default class RoomHeader extends React.Component {
         return !(currentPinEvent.getContent().pinned && currentPinEvent.getContent().pinned.length <= 0);
     }
 
+    _displayInfoDialogAboutScreensharing() {
+        Modal.createDialog(InfoDialog, {
+            title: _t("Screensharing has changed"),
+            description: _t("You don't have to shift-click anymore! You can now share " +
+            "your screen in any video call and in voice calls if other side supports it."),
```
Comment (dbkr) [inline src/components/views/rooms/RoomHeader.js:126]: I'll let design weigh in on this too, but suggest maybe:

> ***Screen sharing is here!***
> 
> You can now share your screen by pressing the 'screen share' button during a call. You can even do this in audio calls if both sides support it!

**Review by dbkr — APPROVED** [review]
Otherwise I think this looks great, apologies for not getting to the review sooner!

File: src/components/views/rooms/RoomHeader.js (hunk 1 above)
Comment (SimonBrandner) [reply src/components/views/rooms/RoomHeader.js]: Yeah, sounds much better!

**Comment by SimonBrandner** [comment]
By a partial accident, I've noticed that screen-sharing designs have been added :smile: Since I had free time, I decided to implement most of them. I hope this makes things easier. In case this isn't desired, I can make a copy of this branch and drop the changes so that we can deal with the new designs in a separate PR.

**Review by gaelledel — CHANGES_REQUESTED** [review]
- [ ] The core of the UI should be to display the screen/tab/window to be shared. Currently it shows the user/black screen.
Screenshot 2021-07-21 at 09 42 50


- [ ] The right side of the UI should be for users videofeeds. Currently, it shows the users but also a PIP of the screen being shared see below
Screenshot 2021-07-21 at 09 36 12

- [ ] When the video feed button is disabled (to not show the video feeds on the right), the top toast indicator string should be updated.

- [ ] Please see the flow of what should happen for the non presenter users here https://www.figma.com/file/V6m2z0oAtUV1l8MdyIrAep/VoIP?node-id=5023%3A162965

- [ ] When users disable their camera, the PIP should still be displayed with the user's avatar. Turning the camera Off shouldn't control whether the PIP window is displayed or not.

**Review by gaelledel — APPROVED** [review]

**Review by dbkr — APPROVED** [review]
🎉 I think let's plan to land this after we cut the next RC.

**Review by dbkr — APPROVED** [review]

**Pull request merged**